includes/Blocks/Block.php

```diff
@@ -286,10 +286,15 @@ private function resolve_block_attributes( $block, $attribute_name, $attribute_c
 
 					break;
 			}//end switch
-
-			// if type is set to integer, get the integer value of the attribute.
-			if ( 'integer' === $attribute_config['type'] ) {
-				$value = intval( $value );
+			// Post processing of return value based on configured type
+			switch ( $attribute_config['type'] ) {
+				case 'integer':
+					$value = intval( $value );
+					break;
+				case 'boolean':
+					// Only false when value is not null or actually false
+					$value = ! ( false === $value || is_null( $value ) );
```

Comment:
Do we need to be concerned with other falsey values like an empty string?

Im not too familiar with the Block Attribute [type validation](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-attributes/#type-validation), but WP is usually horrible at strict booleans.

Reply:
I updated the code to handle null returns now. This should work as expected.